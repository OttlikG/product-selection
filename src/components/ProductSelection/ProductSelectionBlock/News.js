
import React            from 'react'
import Checkbox         from 'components/Atoms/Checkbox'

export default function News (props) {
  const { data, selectProductById } = props

  return (
    <div className="block news-block">
      <div>News</div>

      { data.map(news =>
        <Checkbox
          key={news.id}
          isSelected={news.isSelected}
          label={news.product_name}
          onChange={() => selectProductById(news.id)}
        />)
      }
    </div>
  )
}
