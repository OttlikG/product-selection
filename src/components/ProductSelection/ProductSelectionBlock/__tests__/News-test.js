
import React from 'react'
import News from '../News'

describe('News', () => {
  let props, component, inputs

  const skyNews = {
    id: 'aaa',
    category: 'News',
    product_name: 'Sky News',
    isSelected: true,
  }
  const skySportsNews = {
    id: 'bbb',
    category: 'News',
    product_name: 'Sky Sports News',
    isSelected: false,
  }

  beforeEach(() => {
    props = {
      data: [skyNews, skySportsNews],
      selectProductById: jest.fn(),
    }

    component = render(<News {...props} />)
    inputs = component.findAll('input')
  })

  it('shoud render all item', () => {
    expect(component.find('.news-list').children.length).toBe(2)
  })

  it('should set checked state if isSelected true', () => {
    expect(inputs[0].props.checked).toBe(true)
    expect(inputs[1].props.checked).toBe(false)
  })

  it('should call action on change', () => {
    inputs[0].props.onChange()
    expect(props.selectProductById).toHaveBeenCalledTimes(1)
    expect(props.selectProductById).toHaveBeenCalledWith('aaa')
  })
})
