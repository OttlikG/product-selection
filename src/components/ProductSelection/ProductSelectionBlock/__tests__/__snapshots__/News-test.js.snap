// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News shoud render all item 1`] = `
<div
  className="block news-block"
>
  <div>
    News
  </div>
  <div
    className="news-list"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
         
        Sky News
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
         
        Sky Sports News
      </label>
    </div>
  </div>
</div>
`;

exports[`News should call action on change 1`] = `
<div
  className="block news-block"
>
  <div>
    News
  </div>
  <div
    className="news-list"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
         
        Sky News
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
         
        Sky Sports News
      </label>
    </div>
  </div>
</div>
`;

exports[`News should set checked state if isSelected true 1`] = `
<div
  className="block news-block"
>
  <div>
    News
  </div>
  <div
    className="news-list"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
         
        Sky News
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
         
        Sky Sports News
      </label>
    </div>
  </div>
</div>
`;
